packages/core/src/data-editor/data-editor.tsx

```diff
@@ -1542,9 +1545,13 @@ const DataEditorImpl: React.ForwardRefRenderFunction<DataEditorRef, DataEditorPr
                             height: targetRect.height + 2 * paddingY,
                         };
 
-                        let frozenWidth = 0;
-                        for (let i = 0; i < freezeColumns; i++) {
-                            frozenWidth += columns[i].width;
+                        let frozenLeftWidth = 0;
+                        for (let i = 0; i < freezeLeftColumns; i++) {
+                            frozenLeftWidth += columns[i].width;
+                        }
+                        let frozenRightWidth = 0;
+                        for (let i = columns.length - 1; i >= columns.length - 1 - freezeRightColumns; i--) {
```

Comment:
Should this be `i >= columns.length - freezeRightColumns`? Possible off-by-one error.